packages/babel-traverse/src/scope/index.js

```diff
@@ -906,7 +920,21 @@ export default class Scope {
     do {
       const binding = scope.getOwnBinding(name);
       if (binding) return binding;
-    } while ((scope = scope.parent));
+      if (isScopeInFunctionParameter(scope)) {
+        scope = scope.parent;
+        if (!scope) {
+          break;
+        }
+        const binding2 = scope.getOwnBinding(name);
+        if (binding2 && binding2.kind === "param") {
+          return binding2;
+        }
+      }
+      scope = scope.parent;
+      if (!scope) {
+        break;
+      }
+    } while (true);
```

Comment:
Can we keep this as `while ((scope = scope.parent))`?

Reply:
True, I have refactored this part.